File: atcoder/abc368a_cut.hs
import Data.Char (intToDigit)
import Data.List (intersperse)
-- vicfred
-- https://atcoder.jp/contests/abc368/tasks/abc368_a
-- implementation
main :: IO ()
main = do
  [n, k] <- map read . words <$> getLine :: IO [Int]
  a <- map read . words <$> getLine :: IO [Int]
  putStrLn . unwords . map show $ drop (n - k) a ++ take (n - k) a

File: atcoder/abc368a_cut.rb
# uninhm
# https://atcoder.jp/contests/abc368/tasks/abc368_a
# implementation

n, k = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

(0...n).each do |i|
  print a[(n + i - k) % n], ' '
end
